File: ts/384/303.ts
class NumArray {
  sums: number[]
  constructor(nums: number[]) {
    let sum = 0
    this.sums = [sum]
    for (const n of nums) {
      this.sums.push(sum += n)
    }
  }

  sumRange(left: number, right: number): number {
    return this.sums[right + 1] - this.sums[left]
  }
}

/**
 * Your NumArray object will be instantiated and called as such:
 * var obj = new NumArray(nums)
 * var param_1 = obj.sumRange(left,right)
 */

File: py3/384/303.py
from typing import List

class NumArray:
    def __init__(self, nums: List[int]):
        cur_sum = 0
        self.sums = [0]
        for v in nums:
            cur_sum += v
            self.sums.append(cur_sum)

    def sumRange(self, left: int, right: int) -> int:
        return self.sums[right + 1] - self.sums[left]


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(left,right)